src/reducers/AddSiteReducer.js

```diff
@@ -152,13 +151,13 @@ export default (state = INITIAL_STATE, action) => {
             return {...state, siteReadyForUpload};
 
         case ADD_SITE_SUCCESS:
-            return {...INITIAL_STATE, sitesShouldUpdate: true};
+            return {...INITIAL_STATE};
 
         case ADD_SITE_FAILURE:
             return {...state};
 
         case MAP_IS_INITIALIZING:
-            return {...state, sitesShouldUpdate: false};
```

Comment:
Don't need any more of this annoying logic, now that firebase is keeping us updated on-the-fly with any sites list updates.